### PR TITLE
Committed changes for the varnet related to the "test" mode

### PR DESCRIPTION
**Pull request by sbhadra2020**
Edited the files `data/mri_data.py` and `models/varnet/varnet.py` to make the varnet reconstruction code work in the "test" mode.

### CODE REVIEW

File: models/varnet/varnet.py [hunk 1]
```diff
@@ -301,7 +315,7 @@ def add_model_specific_args(parser):
         return parser
 
 
-def create_trainer(args):
+def create_trainer(args, logger=True):
```
Comment (tullie) [inline models/varnet/varnet.py:318]: Can you just remove the logger argument completely? It's not being used in the function anyway

File: models/varnet/varnet.py [hunk 2]
```diff
@@ -58,13 +58,28 @@ def __call__(self, kspace, target, attrs, fname, slice):
                 mean (float): Mean value used for normalization.
                 std (float): Standard deviation value used for normalization.
         """
-        target = T.to_tensor(target)
+        if target is not None:
+            target = T.to_tensor(target)
+            max_value = attrs['max']
+        else:
+            target = torch.zeros([self.resolution,self.resolution],dtype=torch.float32)
```
Comment (tullie) [inline models/varnet/varnet.py:65]: These target and max_value isn't ever used, right? 

Can we just have:
target = None
max_value = None

Or if the dataloader collate function doesn't work with None (can't remember if it does):
target = torch.tensor(0)
max_value = 0.0

File: models/varnet/varnet.py [hunk 3]
```diff
@@ -254,14 +269,13 @@ def validation_step(self, batch, batch_idx):
         }
 
     def test_step(self, batch, batch_idx):
-        input, _, mean, std, fname, slice = batch
-        output = self.forward(input)
-        mean = mean.unsqueeze(1).unsqueeze(2)
-        std = std.unsqueeze(1).unsqueeze(2)
+        masked_kspace, mask, target, fname, slice, _ = batch
+        output = self.forward(masked_kspace, mask)
+        _, output = T.center_crop_to_smallest(target, output)
```
Comment (tullie) [inline models/varnet/varnet.py:274]: I see now why you make the target args.resolution size. I think it's still better to make the target = None and then here use T.center_crop(args.resolution, args.resolution) to make it clear in the test function that we're not using a target.

File: data/mri_data.py [hunk 4]
```diff
@@ -75,8 +76,9 @@ def __getitem__(self, i):
         fname, slice, padding_left, padding_right = self.examples[i]
         with h5py.File(fname, 'r') as data:
             kspace = data['kspace'][slice]
+            mask = np.asarray(data['mask']) if 'mask' in data else None
             target = data[self.recons_key][slice] if self.recons_key in data else None
             attrs = dict(data.attrs)
             attrs['padding_left'] = padding_left
             attrs['padding_right'] = padding_right
-            return self.transform(kspace, target, attrs, fname.name, slice)
+            return self.transform(kspace, mask, target, attrs, fname.name, slice)
```
Comment (tullie) [inline data/mri_data.py:84]: Can you update the models/unet/train_unet data transform to take this mask parameter in as well? Otherwise I suspect it won't run.

File: models/varnet/varnet.py (hunk 1 above)
Comment (sbhadra2020) [reply models/varnet/varnet.py:318]: Sure, no problem. Yes, I forgot to make the U-Net model compatible as well but will try to edit accordingly. I'll remove the `logger` argument as well.

File: models/varnet/varnet.py (hunk 3 above)
Comment (sbhadra2020) [reply models/varnet/varnet.py:274]: Yes I agree. Will make the edits accordingly.

File: data/mri_data.py (hunk 4 above)
Comment (sbhadra2020) [reply data/mri_data.py:84]: Right, the U-Net's data transform needs to be updated as well.

File: models/varnet/varnet.py (hunk 2 above)
Comment (sbhadra2020) [reply models/varnet/varnet.py:65]: I found that the dataloader collate function was throwing errors with None so I just assigned 0 values.

File: models/varnet/varnet.py (hunk 1 above)
Comment (sbhadra2020) [reply models/varnet/varnet.py]: Added a new commit with the changes

File: models/varnet/varnet.py (hunk 2 above)
Comment (sbhadra2020) [reply models/varnet/varnet.py]: New commit should have 
target = torch.tensor(0)
max_value = 0.0

File: models/varnet/varnet.py (hunk 3 above)
Comment (sbhadra2020) [reply models/varnet/varnet.py]: T.center_crop was added as suggested.

File: data/mri_data.py (hunk 4 above)
Comment (sbhadra2020) [reply data/mri_data.py:84]: DataTransform in train_unet.py has been updated.

**Comment by sbhadra2020** [comment]
Thanks! Happy to help.

I must also add that the fastMRI dataset and associated resources have been very helpful for my research. The `varnet` implementation is also pretty cool and the results are quite impressive.
